### PR TITLE
colexec: cleanup the handling of memory limit of 1 byte

### PR DESCRIPTION
**Pull request by yuzefovich**
For some components (sorters, hash aggregators, hash routers) we divide
up the memory limit into several parts. Previously, if a memory limit of
1 was used, those parts could end up with the limit of zero which could
get treated as "no limit". This is undesirable and is now fixed (note
that other small values of memory limits in theory could still lead to
the old behavior, but it's ok).

The value 1 is special because it is how we implement `ForceDiskSpill`
testing scenario. Still, previously we were exercising the disk spilling
because we would override the zero value to 1 before creating a limited
memory monitor.

In other words, this commit makes several components obey
`distsql_workmem = '1B'` setting (i.e. end to end) and cleans up how the
`ForceDiskSpill` is handled.

Release note: None

### CODE REVIEW

**Review by rharding6373 — APPROVED** [review]
Nice catch! Was this causing many tests to not trigger disk spilling? I'm curious that no tests were failing and we still have no way of knowing (outside of manual investigation, I think) that disk spilling is actually happening in tests. Any ideas on how we could ensure that tests that exercise some broad feature actually exercise the feature?

:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73322#-:-MpnwCaj1QOoaqOYRC4C:b4229ic)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by yuzefovich** [comment]
Ok, it's not as bad as I initially thought - we did override the zero value of limit in `CreateMemAccountForSpillStrategyWithLimit` for both the sorter and the hash aggregator, so we still were exercising the disk spilling logic. For the hash routers we don't use `ForceDiskSpill` knob at all. So this commit only fixes the end-to-end forcing of the disk spilling by `SET distsql_workmem='1B'` (which is how I noticed it), so I think it's still beneficial.

In terms of making sure that we spill to disk when we expect to during the unit tests, we do have some checks for this (we pass in a callback that is called when the spilling to disk occurs), so I think we're good on that front - I was just too quick to jump to conclusions.

Maybe take another quick look at the commit message and the code changes?

**Review by rharding6373 — APPROVED** [review]
Thanks for the clarification! :lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73322#-:-MprA3665tc7cY6A5OP5:b-mcrtbr)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Comment by yuzefovich** [comment]
TFTR!

bors r+
